soilscape_upscaling/upscaling_utilities.py

```diff
@@ -29,92 +28,101 @@ def get_gdal_format(file_name):
-    elif max_value == 0.3:
-        out_colour_table = '''
-            <rsgis:colour name="class_name_1" id="1"  band="{0}" lower="0" upper="0.03" red="165" green="0" blue="38" />
-            <rsgis:colour name="class_name_2" id="2"  band="{0}" lower="0.03" upper="0.06" red="215" green="48" blue="39" />
-            <rsgis:colour name="class_name_3" id="3"  band="{0}" lower="0.06" upper="0.09" red="244" green="109" blue="67" />
-            <rsgis:colour name="class_name_3" id="4"  band="{0}" lower="0.09" upper="0.12" red="253" green="174" blue="97" />
-            <rsgis:colour name="class_name_3" id="5"  band="{0}" lower="0.12" upper="0.15" red="254" green="224" blue="144" />
-            <rsgis:colour name="class_name_3" id="6"  band="{0}" lower="0.15" upper="0.18" red="224" green="243" blue="248" />
-            <rsgis:colour name="class_name_3" id="7"  band="{0}" lower="0.18" upper="0.21" red="171" green="217" blue="233" />
-            <rsgis:colour name="class_name_3" id="8"  band="{0}" lower="0.21" upper="0.24" red="116" green="173" blue="209" />
-            <rsgis:colour name="class_name_3" id="9"  band="{0}" lower="0.24" upper="0.27" red="69" green="117" blue="180" />
-            <rsgis:colour name="class_name_3" id="10" band="{0}" lower="0.27" upper="0.30" red="49" green="54" blue="149" />
-        '''.format(band)
+        # Add class field:
+        bandDefns = []
+        bandDefns.append(BandDefn('SM',inimage,1))
+
+        expression = []
+        expression.append('0.00')
+        expression.append('(SM > 0.00) && (SM <= 0.05)? 1 : 0')
+        expression.append('(SM > 0.05) && (SM <= 0.10)? 2 : 0')
+        expression.append('(SM > 0.10) && (SM <= 0.15)? 3 : 0')
+        expression.append('(SM > 0.15) && (SM <= 0.20)? 4 : 0')
+        expression.append('(SM > 0.20) && (SM <= 0.25)? 5 : 0')
+        expression.append('(SM > 0.25) && (SM <= 0.30)? 6 : 0')
+        expression.append('(SM > 0.30) && (SM <= 0.35)? 7 : 0')
+        expression.append('(SM > 0.35) && (SM <= 0.40)? 8 : 0')
+        expression.append('(SM > 0.40) && (SM <= 0.45)? 9 : 0')
+        expression.append('(SM > 0.45) && (SM <= 0.50)? 10 : 0')
+
+        gdalformat = get_gdal_format(colorimage)
+        datatype = rsgislib.TYPE_8INT
+        color = []
+        for i in range(11):
+           color.append(colorimage.replace('.kea',str(i)+'.kea'))
+           imagecalc.bandMath(color[i], expression[i], gdalformat, datatype, bandDefns)
+
+        bandDefns = []
+        for i in range(11):
+            SMclass = 'SM'+str(i)
+            bandDefns.append(BandDefn(SMclass,color[i],1))
+            if i == 0: 
+                expression = '('+SMclass
+            else:
+                expression = expression+'+'+SMclass
+        expression = expression+')'
+        print('expression: ',expression)
+        imagecalc.bandMath(colorimage, expression, gdalformat, datatype, bandDefns)       
+        for i in range(11):
+            os.system('rm '+color[i])
```

Comment:
Better to use `os.remove` here, should also work under Windows. Alternatively could use a temp directory (`tempfile.mkdtemp`) for the individual colour bands and remove the entire thing using `shutil.rmtree` which would also take care of any  .aux.xml ancillary files.